### PR TITLE
[DA-2515] genomic_set_member history table

### PR DESCRIPTION
**Pull request by kenny-skaggs**
## Resolves *[DA-2515](https://precisionmedicineinitiative.atlassian.net/browse/DA-2515)*
This adds a history table and triggers for the genomic_set_member table. Hooks were also created for alembic so that we can automatically detect changes being made to the genomic_set_member table and also apply them to the history table (as well as update the triggers).

## Description of changes/additions
The alembic changes allow for defining which columns shouldn't be tracked as part of the history table. And any columns that are added or removed from the base table should have revisions generated to keep the history table up to date.

## Tests
- [ ] unit tests

### CODE REVIEW

**Review by dtharpeuno — APPROVED** [review]
👍

File: rdr_service/alembic/env.py [hunk 1]
```diff
@@ -159,6 +159,7 @@ def run_migrations_online():
                 target_metadata=target_metadata.get(name),
                 include_schemas=True,
                 include_object=include_object_fn(name),
+                process_revision_directives=database.AutoHistoryRevisionGenerator.process_revision_directives
```
Comment (j-kanuch) [inline rdr_service/alembic/env.py:162]: Does this process every migration, or just the migrations that are greater than the current revision head?

File: rdr_service/model/genomics.py [hunk 2]
```diff
@@ -54,6 +54,31 @@ class GenomicSetMember(Base):
...
+        'cvl_w2_manifest_job_run_id',
+        'cvl_w3_manifest_job_run_id',
+        'cvl_w4_manifest_job_run_id',
+        'cvl_w4f_manifest_job_run_id',
+        'cvl_aw1c_manifest_job_run_id',
+        'cvl_aw1cf_manifest_job_run_id',
+        'cvl_w3sr_manifest_job_run_id',
+        'cvl_w2sc_manifest_job_run_id',
+        'color_metrics_job_run_id'
+    ]
```
Comment (j-kanuch) [inline rdr_service/model/genomics.py:87]: If we want to add a column to this exclusion list after it was added in a previous release, do we need to create a new migration or will adding it to this list be picked up in the `AutoHistoryRevisionGenerator` run?

File: rdr_service/alembic/env.py (hunk 1 above)
Comment (kenny-skaggs) [reply rdr_service/alembic/env.py:162]: This only processes operations in the migration file being created, it wouldn't look at existing migrations (whatever they may be relative to the current revision head of the database). It essentially lets us look at (and modify) what will be in a migration file that is being autogenerated before the file is written.

File: rdr_service/model/genomics.py (hunk 2 above)
Comment (kenny-skaggs) [reply rdr_service/model/genomics.py:87]: It wouldn't be automatically detected, we would need to create a new migration for it. Do you think that's something that has to be in now, or can that be something we're conscious of and maybe add later if it happens often enough?

File: rdr_service/alembic/env.py (hunk 1 above)
Comment (j-kanuch) [reply rdr_service/alembic/env.py:162]: Sounds good.

File: rdr_service/model/genomics.py (hunk 2 above)
Comment (j-kanuch) [reply rdr_service/model/genomics.py:87]: That should be OK to defer for now. Going forward, if we create a migration for a new field and we want it excluded, we'll add it to the list at that time.

**Review by j-kanuch — APPROVED** [review]

**Review by j-kanuch — APPROVED** [review]

**Pull request merged**